src/glarea.rs

```diff
@@ -0,0 +1,203 @@
+//! # GLArea Sample
+//!
+//! This sample demonstrates how to use GLAreas and OpenGL
+
+extern crate gtk;
+extern crate libc;
+
+// make moving clones into closures more convenient
+macro_rules! clone {
+    ($($n:ident),+; || $body:block) => (
+        {
+            $( let $n = $n.clone(); )+
+            move || { $body }
+        }
+    );
+    ($($n:ident),+; |$($p:ident),+| $body:block) => (
+        {
+            $( let $n = $n.clone(); )+
+            move |$($p),+| { $body }
+        }
+    );
+}
+
+#[cfg(feature = "opengl")]
+extern crate epoxy;
+
+// Based off of the static-triangle example from gl-rs
+#[cfg(feature = "opengl")]
+mod example {
+    use std::ffi;
+    use std::mem;
+    use std::ptr;
+
+    use gtk;
+    use gtk::traits::*;
+    use gtk::signal::Inhibit;
+    use gtk::{GLArea, Window};
+
+    use epoxy;
+    use epoxy::types::*;
+    use epoxy::Gl;
+
+    fn compile_shader(src: &str, ty: GLenum) -> GLuint {
+        unsafe {
+            let shader = Gl.CreateShader(ty);
+            // Attempt to compile the shader
+            let csrc = ffi::CString::new(src).unwrap().as_ptr();
+            Gl.ShaderSource(shader, 1, &csrc, ptr::null());
```

Comment:
Hmm, it seems you're not required to use `CString` here because the function takes an array of lengths?
I'm somewhat allergic to `CString` because it's a popular source of accidental dangling pointers (not that you did anything wrong here).